presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/type/AbstractPrestoTimestampEncoder.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.base.type;
+
+import io.prestosql.spi.block.BlockBuilder;
+import io.prestosql.spi.type.TimestampType;
+import org.joda.time.DateTimeZone;
+
+abstract class AbstractPrestoTimestampEncoder<T extends Comparable<T>>
+        implements PrestoTimestampEncoder<T>
+{
+    protected final DateTimeZone timeZone;
+    protected final TimestampType type;
+
+    AbstractPrestoTimestampEncoder(TimestampType type, DateTimeZone timeZone)
+    {
+        this.type = type;
+        this.timeZone = timeZone;
+    }
+
+    @Override
+    public T write(DecodedTimestamp decodedTimestamp, BlockBuilder blockBuilder)
+    {
+        T timestamp = getTimestamp(decodedTimestamp);
+        write(timestamp, blockBuilder);
+        return timestamp;
+    }
+
+    @Override
+    public TimestampType getType()
+    {
+        return type;
+    }
+
+    protected abstract T write(T timestamp, BlockBuilder blockBuilder);
```

Comment:
This doesn't make the code simple.
Remove the default implementation of `write(DecodedTimestamp decodedTimestamp, BlockBuilder blockBuilder)` above. If a subclass chooses to reuse `getTimestamp`, it still can do it.